pkg/docker/config/config_test.go

```diff
@@ -216,10 +239,26 @@ func TestGetAuth(t *testing.T) {
 				if tc.sys != nil {
 					sys = tc.sys
 				}
-				username, password, err := GetAuthentication(sys, tc.hostname)
+				auth, err := GetCredentials(sys, tc.hostname)
 				assert.Equal(t, tc.expectedError, err)
-				assert.Equal(t, tc.expectedUsername, username)
-				assert.Equal(t, tc.expectedPassword, password)
+				assert.Equal(t, tc.expected.Username, auth.Username)
+				assert.Equal(t, tc.expected.Password, auth.Password)
+				assert.Equal(t, tc.expected.IdentityToken, auth.IdentityToken)
+
+				// Test for the previous APIs.
+				username, password, err := GetAuthentication(sys, tc.hostname)
+				switch err {
+				case nil:
+					assert.Equal(t, tc.expected.Username, username)
+					assert.Equal(t, tc.expected.Password, password)
```

Comment:
`assert.empty(t, tc.expected.IdentityToken)` ?

Reply:
More directly, the switch could be based on `tc.expected.IdentityToken`  / `tc.expectedError` , so that the expectations drive what is being compared, instead of return values driving what the expectations “must” have been.